modules/dag-l1/src/main/scala/org/tessellation/dag/l1/domain/block/BlockService.scala

```diff
@@ -67,6 +59,24 @@ object BlockService {
 
         def getCollateral: Amount = collateral
       }
+
+      private def processAcceptanceSuccess(hashedBlock: Hashed[DAGBlock])(contextUpdate: BlockAcceptanceContextUpdate): F[Unit] = for {
+        hashedTransactions <- hashedBlock.signed.transactions.toNonEmptyList
+          .sortBy(_.ordinal)
+          .traverse(_.toHashed)
+
+        _ <- hashedTransactions.traverse(transactionStorage.accept)
+        _ <- blockStorage.accept(hashedBlock)
+        _ <- addressStorage.updateBalances(contextUpdate.balances)
+        isDependent = (block: Signed[DAGBlock]) => BlockRelations.dependsOn(hashedBlock)(block)
+        _ <- blockStorage.restoreDependent(isDependent)
+      } yield ()
+
+      private def processAcceptanceError(hashedBlock: Hashed[DAGBlock])(reason: BlockNotAcceptedReason): F[BlockAcceptanceError] =
+        blockStorage
+          .postpone(hashedBlock)
+          .as(BlockAcceptanceError(BlockReference(hashedBlock.height, hashedBlock.proofsHash), reason))
```

Comment:
Hmm, I saw `ownReference` in `HashedOps`. isn't it that case?

Reply:
True, there is `ownReference`. So I guess it can be used here, but we could also move the functionality to the `of` function given that we already have `of` for `Signed[DAGBlock]`. Would be best to use one approach.